BGAnimations/ScreenSelectMusic overlay/PerPlayer/DensityGraph.lua

```diff
@@ -0,0 +1,137 @@
+local player = ...
+local pn = ToEnumShortString(player)
+local p = PlayerNumber:Reverse()[player]
+
+local height = 64
+local width = IsUsingWideScreen() and 286 or 276
+
+-- Max height at 280 BPM 16ths. Anything faster will just scale the graph. 
+local maxNps = 280/60 * 4
+
+local af = Def.ActorFrame{
+	InitCommand=function(self)
+		self:visible( GAMESTATE:IsHumanPlayer(player) )
+		self:xy(_screen.cx-182, _screen.cy+23)
+
+		if player == PLAYER_2 then
+			self:addy(height+24)
+		end
+
+		if IsUsingWideScreen() then
+			self:addx(-5)
+		end
+	end,
+	PlayerJoinedMessageCommand=function(self, params)
+		if params.Player == player then
+			self:visible(true)
+		end
+	end,
+	PlayerUnjoinedMessageCommand=function(self, params)
+		if params.Player == player then
+			self:visible(false)
+		end
+	end,
+
+	CurrentSongChangedMessageCommand=function(self) self:queuecommand("UpdateGraph") end,
+	CurrentCourseChangedMessageCommand=function(self) self:queuecommand("UpdateGraph") end,
```

Comment:
Maybe not required because of the StepsP*Changed commands? (not tested)

Reply:
Looks like both of those are indeed not required. Removed.